Visitor can see different products on a restaurants menu

yasmineezequiel opened the pull request:
[PT-link](https://www.pivotaltracker.com/story/show/168829429)
Yasmine & Pia

# PR contains
In this PR we've created a menu index-page where we display different products from a restaurants menu. 

## This PR contains
- Feature-test
  - with DB-table for pre-existing products
- Menu controller
- Menu index page
- Product model 
  - with name, description and price
- FactoryBot
- Product spec tests
- List with content on menu index page

## What we've learned
- Learned how to convert code in .erb into .haml
- Deeper understanding of working with BDD. 

<img width="637" alt="Screenshot 2019-10-01 at 10 43 45" src="https://user-images.githubusercontent.com/52795566/65947344-5f51ab00-e438-11e9-92be-92925c9d8db3.png">

![20190930_135343_001](https://user-images.githubusercontent.com/52795566/65902860-97ff6f00-e3bb-11e9-9bf5-4cb9b5bc6c45.jpg)

yasmineezequiel committed:
Refactors erb-files to haml-files.

yasmineezequiel committed:
Adds Feature Scenario for User visiting the webpage

yasmineezequiel committed:
Updates feature-scenario and basic-steps.

yasmineezequiel committed:
Creates menu-controller, adds content to menu-index.

yasmineezequiel committed:
Adds feature test for visiting the page

yasmineezequiel committed:
Adds menu in idex file

faraznaeem commented on config/routes.rb:
```diff
@@ -1,3 +1,7 @@
+
+get 'menu/index'
+
+root 'menu#index'
```
Please use the proper syntax when writing routes, and make sure that you use resourceful routing.

faraznaeem reviewed (changes requested):
This feature seems incomplete, the story states that we should see products on the resturants menu. However, this feature only has hardcoded information on the landing page. Please add the Product model and review this feature again to meet the requirements of the story.

yasmineezequiel committed:
Adds feature background for pre-existing products.

Co-authored-by: Pia von Wachenfelt <piavonwachenfelt@hotmail.com>

yasmineezequiel committed:
Adds product model datatable
- Creates product FactoryBot
- Creates product spec-file
- Adds validations
>
>
Co-authored-by: Pia von Wachenfelt <piavonwachenfelt@hotmail.com>

yasmineezequiel committed:
Refactors haml-code, adds background and scenario step.

Co-authored-by: Pia von Wachenfelt <piavonwachenfelt@hotmail.com>

yasmineezequiel committed:
Refactors index and steps

Co-authored-by: Pia von Wachenfelt <piavonwachenfelt@hotmail.com>

clalil reviewed (changes requested):
- Remove all br
- Change hash in basic_steps to product.

jonas-masharqa reviewed (approved):
Looks good!

faraznaeem commented on app/models/product.rb:
```diff
@@ -0,0 +1,5 @@
+class Product < ApplicationRecord
+  validates :name, presence: true
+  validates :description, presence: true
+  validates :price, presence: true
```
Refactor the validations

faraznaeem commented on config/routes.rb:
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  get 'menu/index'
```
as we say in swedish: aja baja

faraznaeem commented on features/support/menu.feature:
```diff
@@ -0,0 +1,20 @@
+  Scenario: Be able to see menu
+    Given I visit the page 
+    Then I should see 'Menu'
+    And I should see "garlic bread"
```
You should test all of the products and their attributes in the background and not only garlic bread

yasmineezequiel committed:
Adds feature test scenario

Co-authored-by: Pia von Wachenfelt <piavonwachenfelt@hotmail.com>

yasmineezequiel committed:
Adds products to seeds

Co-authored-by: Pia von Wachenfelt <piavonwachenfelt@hotmail.com>

yasmineezequiel committed:
[Debuggning - semaphore] Updates migrate and moves code in routes.

Co-authored-by: Pia von Wachenfelt <piavonwachenfelt@hotmail.com>

yasmineezequiel committed:
Re-name controllers and view to product

oliverochman commented on features/basic_steps.rb:
```diff
@@ -0,0 +1,13 @@
+  visit root_path
+end
+
+Then("I should see {string}") do |string|
```
this is an assertion, please put it in an assertion_steps file

oliverochman commented on features/support/menu.feature:
```diff
@@ -0,0 +1,25 @@
+Feature: Visitor can view menu
+As a visitor 
```
line 2 to 4 should be indented

oliverochman commented on db/seeds.rb:
```diff
@@ -1,7 +1,9 @@
-#
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+product = Product.create(name: "Garlic Bread", description: "garlic, bread, butter", price: 30) 
```
products = [
  Product.create(name: "Garlic Bread", description: "garlic, bread, butter", price: 30) ,
  Product.create(name: "Margherita", description: "cheese, tomato sauce, basil", price: 70),
  Product.create(name: "Vesuvio", description: "cheese, tomato sauce, ham", price: 70),
  Product.create(name: "Calzone", description: "cheese, tomato sauce, ham", price: 80),
  Product.create(name: "Ice-cream", description: "vanilla", price: 40),
  Product.create(name: "Ice-cream", description: "strawberry", price: 40)
]

yasmineezequiel committed:
Creats assertion_steps and refactors scenario and seeds

Co-authored-by: Pia von Wachenfelt <piavonwachenfelt@hotmail.com>

oliverochman reviewed (approved):
lgtm!
